_posts/2016-03-16-AngularTS.md

```diff
@@ -0,0 +1,413 @@
+---
+layout: post
+author: kevin_van_houtte
+title: 'AngularTS: A new look @ Angular'
+image: /img/angularts.jpg
+tags: [TypeScript, AngularJS]
+category: AngularJS, TypeScript
+comments: true
+---
+
+## Combining the best of two worlds.
+
+Since my introduction to the heroic AngularJS framework at Devoxx around 4 years ago.
+I was intrigued and set for an adventure.
+With Angular 2 release in sight we have to prepare ourselves with the migrating road map coming up.
+One of the core changes in angular, was changing the code language to TypeScript.
+This blog will cover the use of angular components in TypeScript.
+But what is this TypeScript? 
+TypeScript is a superset of JavaScript that focuses on strong typing and the new ES6 features: classes, interfaces and modules.
+Like in common OO languages like Java and C# these features aren’t new.
+These features gives the developer the opportunity to build a object oriented architecture in JavaScript.
+With that in mind let’s see what the advantages are: 
+
+### Transpiling
+The DOM on your website can only recognize JavaScript.
+With this said they had to come with a way to compile TS.
+Because TS is a superset of JS it can transpile to plain JavaScript before including it into HTML.
+Transpilers are integrated in the latest IDE’s. 
+Any valid JavaScript is valid TypeScript.
+
+### Strongly typed
+When you are used to plain JavaScript, you notice that every time you need a variable, your type is loosely typed.
+With TypeScript they give you the opportunity to give every variable its own type.
+This comes with great beneficials like better refactoring, less bugs and better type checking at compile time.
+
+### OO architecture
+TypeScript offers a object oriented architecture experience, which means all code is defined in classes, interfaces and most of those classes can be instantiated into objects. It also supports encapsulation, which protects the data from unintended access and modification. 
+
+
+## Learning path AngularTS
+If you’re no stranger to AngularJS you will notice that the structure remains the same. 
+Two way databinding, controllers, services, ... 
+But be aware that it has a different syntax in TypeScript. 
+I will show you the different best practices to implement these components. 
+
+
+## TypeScript Definition Files
+When using TS we have to reference to TSD files, these files describe the types defined in external libraries such as angular. 
+To install the angular tsd files we use the tsd manager.
+To use the tsd manager we install it with:
+{% highlight bash %}
+npm install tsd -g
+{% endhighlight %}
+
+Afterwards install angular with:
```

Comment:
Would put the explanation parts outside the code block and use backticks (\`) when referring to the options:
`--resolve` will include all dependencies angular needs.
`--save` will include the angular tsd into the tsd.json file.